builder/processor/src/main/java/io/helidon/builder/processor/TypeHandlerSupplier.java

```diff
@@ -0,0 +1,190 @@
+                    .addLine("return self();"));
+        }
+
+        if (factoryMethod.builder().isPresent()) {
+            // if there is a factory method for the return type, we also have setters for the type (probably config object)
+            FactoryMethods.FactoryMethod fm = factoryMethod.builder().get();
+
+            TypeName builderType;
+            String className = fm.factoryMethodReturnType().className();
+            if (className.equals("Builder") || className.endsWith(".Builder")) {
+                builderType = fm.factoryMethodReturnType();
+            } else {
+                builderType = TypeName.create(fm.factoryMethodReturnType().fqName() + ".Builder");
+            }
+            String argumentName = "consumer";
+            TypeName argumentType = TypeName.builder()
+                    .type(Consumer.class)
+                    .addTypeArgument(builderType)
+                    .build();
+
+            classBuilder.addMethod(builder -> builder.name(setterName())
+                    .accessModifier(setterAccessModifier(configured))
+                    .description(blueprintJavadoc.content())
+                    .returnType(returnType, "updated builder instance")
+                    .addParameter(param -> param.name(argumentName)
+                            .type(argumentType)
+                            .description(blueprintJavadoc.returnDescription()))
+                    .addJavadocTag("see", "#" + getterName() + "()")
+                    .typeName(Objects.class)
+                    .addLine(".requireNonNull(" + argumentName + ");")
+                    .add("var builder = ")
+                    .typeName(fm.typeWithFactoryMethod().genericTypeName())
+                    .addLine("." + fm.createMethodName() + "();")
+                    .addLine("consumer.accept(builder);")
+                    .addLine("this." + name() + "(builder.build());")
+                    .addLine("return self();"));
+        }
+    }
+
+    protected void declaredSetter(InnerClass.Builder classBuilder,
+                                  PrototypeProperty.ConfiguredOption configured,
+                                  TypeName returnType,
+                                  Javadoc blueprintJavadoc) {
+        Method.Builder builder = Method.builder()
+                .name(setterName())
+                .returnType(returnType, "updated builder instance")
+                .description(blueprintJavadoc.content())
+                .addJavadocTag("see", "#" + getterName() + "()")
+                .addParameter(param -> param.name(name())
+                        .type(argumentTypeName())
+                        .description(blueprintJavadoc.returnDescription()))
+                .accessModifier(setterAccessModifier(configured));
+        builder.typeName(Objects.class).addLine(".requireNonNull(" + name() + ");");
+        builder.addLine("this." + name() + " = " + name() + "::get;")
+                .addLine("return self();");
```

Comment:
Since these are changes on the same method builder, why not to just continue with the method chain? Also `.typeName(Objects.class).addLine(".requireNonNull(" + name() + ");")` is in above method body creations on the new line and here it is on the same. I would say it should be on the new line since it is chained method call.

Reply:
Fixed